pyproject.toml

```diff
@@ -1,6 +1,6 @@
 [project]
 name = "tensorizer"
-version = "1.1.1"
+version = "1.2.0"
```

Comment:
This currently contains breaking changes on the code and data sides of the library, so if we're following semantic versioning, we ought to bump this to v2.0.0 with this PR.

2/3 of the significant breaking changes could be made compatible by:
- Keeping the `_convert_dtype_to_torch` function to allow `numpy` dtype arguments for the deserializer, where valid, and
- Writing `TENSORIZER_VERSION` as 1 in compatible serialized files (as discussed in other comments).

The only one that seems mostly unavoidable is altering the return value type of `TensorDeserializer.read_tensors()` from `numpy.ndarray`s to `torch.Tensor`s (at least when opaque dtypes are present). `read_tensors()` returning tensors is more sensible anyway, but it does change the interface a small bit.

Reply:
From a discussion with @wbrown on slack, there is still user desire to deal with numpy as input/output types. The serializer already does this by `write_tensors` allowing both numpy ndarrays and torch tensors as input. To keep the same "single function, multiple in/outputs" pattern, I'll add a `as_numpy`  flag to `read_tensors()` that controls what dtype is returned.

This doesn't change the fact that this PR changes the default behavior and is therefore a breaking change, but it still gives a path for the user to interact with tensorizer only through numpy ndarray's.

They won't be able to use the "global dtype" in the deserializer's constructor to cast the numpy arrays but that was always only done when registering it to the torch module.